src/melee.h

```diff
@@ -2,11 +2,26 @@
 #ifndef CATA_SRC_MELEE_H
 #define CATA_SRC_MELEE_H
 
+/*
+ * statistics data for melee attacks, used for test purposes
+ */
+struct melee_statistic_data {
```

Comment:
These melee stats are a good candidate to migrate to the event bus, then the tests can hook in to the hit events that way.  I'm not asking you to do that in this PR, because melee events aren't a thing yet; this is more of a note-to-self.

Reply:
Sounds great =)